frontend/src/metabase/query_builder/components/dataref/ModelPane/ModelPane.jsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { connect } from "react-redux";
+import { t, jt } from "ttag";
+import { createSelector } from "reselect";
+import PropTypes from "prop-types";
+
+import { getMetadata } from "metabase/selectors/metadata";
+import MetabaseSettings from "metabase/lib/settings";
+import { formatDateTimeWithUnit } from "metabase/lib/formatting";
+import {
+  Description,
+  EmptyDescription,
+} from "metabase/components/MetadataInfo/MetadataInfo.styled";
+import Question from "metabase-lib/lib/Question";
+import FieldList from "../FieldList";
+import {
+  ModelPaneDetail,
+  ModelPaneDetailLink,
+  ModelPaneDetailText,
+  ModelPaneIcon,
+  ModelPaneDescription,
+} from "./ModelPane.styled";
+
+// This formats a timestamp as a date using any custom formatting options.
+function formatDate(value) {
+  const options = MetabaseSettings.get("custom-formatting")["type/Temporal"];
+  return formatDateTimeWithUnit(value, "day", options);
+}
+
+const getQuestion = createSelector(
+  [getMetadata, (_state, card) => card],
+  (metadata, card) => new Question(card, metadata),
+);
+
+const mapStateToProps = (state, props) => ({
+  question: getQuestion(state, props.model),
+});
+
+const propTypes = {
+  show: PropTypes.func.isRequired,
+  model: PropTypes.object.isRequired,
+  question: PropTypes.object.isRequired,
+};
+
+const ModelPane = ({ show, model, question }) => {
```

Comment:
Also not clear why it's JS and not TS

Reply:
I started by creating this file from TablePane.jsx, so it was easier. What are the general rules around choosing between TS vs JS?

Reply:
We use TS for all new files unless there are some outstanding issues